### PR TITLE
Dockerfile.ubi: use tensorizer

### PR DESCRIPTION
**Pull request by prashantgupta24**
add libsodium for tensorizer encryption

### CODE REVIEW

File: Dockerfile.ubi [hunk 1]
```diff
@@ -125,6 +125,22 @@ RUN --mount=type=cache,target=/root/.cache/ccache \
...
 
+#################### libsodium Build IMAGE ####################
+FROM base as libsodium-builder
+
+RUN microdnf install -y gcc gzip \
+    && microdnf clean all
+
+WORKDIR /usr/src/libsodium
+
+ARG LIBSODIUM_VERSION=1.0.19
```
Comment (dtrifiro) [inline Dockerfile.ubi:136]: Is there a specific reason for using `1.0.19` instead of `1.0.20`?

File: Dockerfile.ubi [hunk 2]
```diff
@@ -125,6 +125,22 @@ RUN --mount=type=cache,target=/root/.cache/ccache \
...
+RUN curl -LO https://github.com/jedisct1/libsodium/releases/download/${LIBSODIUM_VERSION}-RELEASE/libsodium-${LIBSODIUM_VERSION}.tar.gz \
+    && tar -xzvf libsodium*.tar.gz \
+    && rm -f libsodium*.tar.gz \
+    && mv libsodium*/* ./
+
+RUN ./configure && make && make check
```
Comment (dtrifiro) [inline Dockerfile.ubi:142]:
```suggestion
RUN ./configure --prefix="/usr/" && make && make check
```

File: Dockerfile.ubi [hunk 3]
```diff
@@ -143,7 +159,13 @@ COPY extras/custom_cache_manager.py /opt/vllm/lib/python3.11/site-packages/custo
...
 RUN --mount=type=bind,from=build,src=/workspace/dist,target=/workspace/dist \
     --mount=type=cache,target=/root/.cache/pip \
-    pip install dist/*.whl --verbose
+    pip install $(echo dist/*.whl)'[tensorizer]' --verbose
+
+# Install libsodium for Tensorizer encryption
+RUN --mount=type=bind,from=libsodium-builder,src=/usr/src/libsodium,target=/usr/src/libsodium \
+    cd /usr/src/libsodium \
+    && make install
+ENV LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH
```
Comment (dtrifiro) [inline Dockerfile.ubi:168]:
```suggestion
```

File: Dockerfile.ubi (hunk 1 above)
Comment (prashantgupta24) [reply Dockerfile.ubi:136]: Uhh @tjohnson31415 did you explore this? I see the `1.0.20` was released before you created the PR in ibm

Comment (tjohnson31415) [reply Dockerfile.ubi:136]: Should be fine to upgrade. Initial integration testing I did was before 1.0.20 was released.

Comment (prashantgupta24) [reply Dockerfile.ubi:136]: Tried an image with `1.0.20`, the integration test works fine.

**Pull request merged**